src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -1349,6 +1357,12 @@ public SyntaxNode GetRightSideOfDot(SyntaxNode node)
                 (node as MemberAccessExpressionSyntax)?.Name;
         }
 
+        public bool IsNodeOrAnyAncestorLeftSideOfDot(SyntaxNode node)
+            => IsNodeOrAnyAncestorLeftSideOfDot(node, this);
```

Comment:
not really a syntax fact either.  the goal of syntaxfacts is not for features to place their algorithms in one place. it's to answer very specific and *direct* questions of syntax in a language neutral fashion.  i.e. "Are you a parenthesized expression".  "get me the parts of this member access expressoin".

Reply:
There is already a method here `IsLeftSideOfDot` and this one is extending it to ask if that is true for given node or any of its ancestor. I feel this seems equally reasonable syntactic query as `IsLeftSideOfDot`.

Reply:
We try not to have algorithms in SyntaxFacts.  Someone who wants to walk up teh ancestors can do that themselves.  You could have an extension here, but i'm also pretty loathe to do that.

Specifically, i do not want ISyntaxFactsService to just be a dumping ground for "i want some code that runs over VB and C#" :)  That's not its purpose (even though about one in five PRs tries to do that).  Instead, it's purpose is to answer very direct questions that match the direct questions you have on *Syntax* nodes already, just in a cross language fashion.

So, for example, there is no "CSharp.SyntaxNode.IsNodeOrAnyAncestorLeftSideOfDot".  

Note that i haven't been able to keep watch of this the entire time, and things have crept in.  But really, the more that creeps in, the more this just becomes: let me just write my feature in this one service that now does everything.

That's not good design, it's not healthy, and we should strive to avoid it.

Reply:
Sounds reasonable, I can refactor this into a separate service.

Reply:
Note: do we actually need this as a separate service?  this just seems like it would be in the core code that is classifying references.

Reply:
I was trying to keep this as an extension method that other features can use in future if needed - but makes sense to keep this in the AbstractReferenceFinder for now, and refactor later if such a client arises.